### PR TITLE
Support for piecewise split of Job result

### PR DESCRIPTION
**Pull request by mdonaka**
depens on https://github.com/acompany-develop/QuickMPC/pull/4

# Summary
split job result

# Purpose
Successfully save to DB and return to Client when job result is huge.

# Contents
- Change return of lib to MC get_computation_result to stream
- Add ability to combine results in piece order when executing get_computation_result
- Change send_model_param to send split into pieces

# Testing Methods Performed
- pytest
- Send a request to get a join table job with larger data

### CODE REVIEW

File: quickmpc/utils/make_pieces.py [hunk 1]
```diff
@@ -30,6 +30,23 @@ def __get_byte_str(s: str):
...
+        if max_size > upper_limit_size:
+            raise RuntimeError(
+                f"max size of piece ({max_size}) is over "
+                f"specified limit size ({upper_limit_size})"
+            )
+        if max_size < lower_limit_size:
+            raise RuntimeError(
+                f"max size of piece ({max_size}) is under "
+                f"specified limit size ({lower_limit_size})"
+            )
```
Comment (come2ry) [inline quickmpc/utils/make_pieces.py:48]: As for here, ok!

**Review by arukuka — APPROVED** [review]

File: tests/unit_tests/test_make_pieces.py [hunk 2]
```diff
@@ -21,15 +22,55 @@ def test_make_pieces(self, matrix, size, expected):
         actual = MakePiece.make_pieces(matrix, size)
         assert(actual == expected)
 
+    @pytest.mark.parametrize(
+        ("string, size,expected"),
```
Comment (come2ry) [inline tests/unit_tests/test_make_pieces.py]:
```suggestion
        ("string, size, expected"),
```

**Review by come2ry — CHANGES_REQUESTED** [review]

File: tests/unit_tests/test_make_pieces.py (hunk 2 above)
Comment (mdonaka) [reply tests/unit_tests/test_make_pieces.py]: thx!

File: tests/unit_tests/test_make_pieces.py [hunk 3]
```diff
@@ -21,15 +22,55 @@ def test_make_pieces(self, matrix, size, expected):
...
+            ("aaabbbcccddd", 9, ["aaabbbccc", "ddd"]),
+            ("aaabbbcccddd", 12, ["aaabbbcccddd"]),
+            ("aaabbbcccddd", 15, ["aaabbbcccddd"]),
+            # 長めの文字列
+            ("".join(["a" for _ in range(1_000_000)]), 1,
+             ["".join(["a" for _ in range(1)]) for _ in range(1_000_000)]),
+            ("".join(["a" for _ in range(1_000_000)]), 1_000,
+             ["".join(["a" for _ in range(1_000)]) for _ in range(1_000)]),
+            ("".join(["a" for _ in range(1_000_000)]), 1_000_000,
+             ["".join(["a" for _ in range(1_000_000)]) for _ in range(1)]),
```
Comment (come2ry) [inline tests/unit_tests/test_make_pieces.py]:
```suggestion
            ('a'*1_000_000, 1,
             ['a'*1]*1_000_000),
            ('a'*1_000_000, 1_000,
             ['a'*1_000]*1_000),
            ('a'*1_000_000, 1_000_000,
             ['a'*1_000_000]*1),
```

**Review by come2ry — CHANGES_REQUESTED** [review]

File: tests/unit_tests/test_make_pieces.py (hunk 3 above)
Comment (come2ry) [reply tests/unit_tests/test_make_pieces.py]: [IMO]
In terms of readability

Comment (mdonaka) [reply tests/unit_tests/test_make_pieces.py]: This is so beautiful. Thank you!
I did some rearranging and fixed it with 74c0a11424276159c651ef662c47d026a345291a

**Review by come2ry — APPROVED** [review]

**Pull request merged**